### DISCUSSION TITLE
Share performance profiling

### DISCUSSION THREAD

**Discussion by njzjz** [deepmodeling/deepmd-kit#942] [Show and tell]
I just used TensorFlow's [new profiling tool](https://www.tensorflow.org/guide/profiler) (`tf.profiler.experimental.Profile`) on the DeePMD-kit. I tested on `examples/water/se_e2_a`. Here's results:

# overall
![image](https://user-images.githubusercontent.com/9496702/128777064-6923c97e-5fdf-4647-903a-42e785e48b7f.png)
![image](https://user-images.githubusercontent.com/9496702/128777897-62747902-38a0-481d-afd0-23a96bcc734c.png)

Average Step Time (lower is better) (σ = 19.4 ms) 19.0 ms
Kernel Launch Time (σ = 4.3 ms) 2.6 ms
Device Compute Time (σ = 16.5 ms) 16.4 ms

![image](https://user-images.githubusercontent.com/9496702/128777375-4df4811a-af14-49f9-987e-7ff3798f2761.png)
![image](https://user-images.githubusercontent.com/9496702/128777440-2e5c0908-8c77-43b6-a21c-0db985556536.png)
![image](https://user-images.githubusercontent.com/9496702/128777496-71f5e9cf-4119-4a6a-b5a7-8a141d2b668e.png)

# GPU Kernel Stats
[data.txt](https://github.com/deepmodeling/deepmd-kit/files/6957564/data.txt)
![image](https://user-images.githubusercontent.com/9496702/128777715-9bf1a974-6411-442d-bb90-d9651d340cdd.png)

# Memory 
![image](https://user-images.githubusercontent.com/9496702/128777788-dd1fe897-7396-4272-acc6-38f4d1666b58.png)
![image](https://user-images.githubusercontent.com/9496702/128777811-f88ca9a1-5e84-4f8f-b9d9-3c02ea3c0fb9.png)

# TF stats
![image](https://user-images.githubusercontent.com/9496702/128777918-dacf815d-3bb9-42bb-83b3-70a625b0a130.png)
![image](https://user-images.githubusercontent.com/9496702/128777946-e709cae1-e40a-449b-bbef-eaa47a001bcf.png)
![image](https://user-images.githubusercontent.com/9496702/128786482-90c0f5c3-6592-47e9-826f-bb44d33faabf.png)

# Trace viewer
![image](https://user-images.githubusercontent.com/9496702/128778065-36799629-af36-4530-b3bd-73ca03b1cd3d.png)
![image](https://user-images.githubusercontent.com/9496702/128778369-6388f74c-a70f-42dc-87ff-c9cbc607a070.png)

**Comment by amcadmus** [comment]
@denghuilu @iProzd @pkulzy Please take a look
